integration/src/integration/java/pl/allegro/tech/hermes/integration/auth/TopicAuthorisationTest.java

```diff
@@ -0,0 +1,206 @@
+import pl.allegro.tech.hermes.test.helper.config.MutableConfigFactory;
+import pl.allegro.tech.hermes.test.helper.endpoint.HermesPublisher;
+import pl.allegro.tech.hermes.test.helper.message.TestMessage;
+import pl.allegro.tech.hermes.test.helper.util.Ports;
+
+import javax.ws.rs.core.Response;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static javax.ws.rs.core.Response.Status.Family.SUCCESSFUL;
+import static pl.allegro.tech.hermes.integration.auth.SingleUserAwareIdentityManager.getHeadersWithAuthentication;
+import static pl.allegro.tech.hermes.integration.test.HermesAssertions.assertThat;
+
+public class TopicAuthorisationTest extends IntegrationTest {
+
+    public static final int FRONTEND_PORT = Ports.nextAvailable();
+    public static final String FRONTEND_URL = "http://127.0.0.1:" + FRONTEND_PORT;
+
+    private static final String USERNAME = "someUser";
+    private static final String PASSWORD = "somePassword123";
+    private static final String MESSAGE = TestMessage.of("hello", "world").body();
+
+    protected HermesPublisher publisher;
+
+    private HermesFrontend hermesFrontend;
+
+    @BeforeClass
+    public void setup() throws Exception {
+        ConfigFactory configFactory = new MutableConfigFactory()
+                .overrideProperty(Configs.FRONTEND_PORT, FRONTEND_PORT)
+                .overrideProperty(Configs.FRONTEND_SSL_ENABLED, false)
+                .overrideProperty(Configs.FRONTEND_AUTHENTICATION_MODE, "pro_active")
+                .overrideProperty(Configs.FRONTEND_AUTHENTICATION_ENABLED, true);
+
+        AuthenticationConfiguration authConfig = new AuthenticationConfiguration(
+                exchange -> false,
+                Lists.newArrayList(new BasicAuthenticationMechanism("basicAuthRealm")),
+                new SingleUserAwareIdentityManager(USERNAME, PASSWORD));
+
+        hermesFrontend = HermesFrontend.frontend()
+                .withBinding(configFactory, ConfigFactory.class)
+                .withAuthenticationConfiguration(authConfig)
+                .build();
+
+        hermesFrontend.start();
+
+        publisher = new HermesPublisher(FRONTEND_URL);
+
+        operations.buildTopic("someGroup", "topicWithAuthorization");
+    }
+
+    @AfterClass
+    public void tearDown() throws InterruptedException {
+        hermesFrontend.stop();
+    }
+
+    @Test
+    public void shouldPublishWhenAuthenticated() {
+        // given
+        List<Topic> topics = Arrays.asList(
+                TopicBuilder.topic("disabled.authenticated")
+                        .build(),
+                TopicBuilder.topic("enabled.authenticated_1Publisher")
+                        .withPublisher(USERNAME)
+                        .withAuthEnabled()
+                        .build(),
+                TopicBuilder.topic("enabled.authenticated_2Publishers")
+                        .withPublisher(USERNAME)
+                        .withPublisher("foobar")
```

Comment:
\[PMD\] INFO: The String literal "foobar" appears 6 times in this file; the first occurrence is on line 87